daft/daft.pyi

```diff
@@ -70,4 +73,618 @@ class ImageFormat(Enum):
+        """Take a field-wise max of the list of resource requests."""
+        ...
+    def __add__(self, other: ResourceRequest) -> ResourceRequest: ...
+    def __repr__(self) -> str: ...
+    def __eq__(self, other: ResourceRequest) -> bool: ...  # type: ignore[override]
+    def __ne__(self, other: ResourceRequest) -> bool: ...  # type: ignore[override]
+
+class FileFormat(Enum):
+    """
+    Format of a file, e.g. Parquet, CSV, and JSON.
+    """
+
+    Parquet: int
+    Csv: int
+    Json: int
+
+class ParquetSourceConfig:
+    """
+    Configuration of a Parquet data source.
+    """
+
+class CsvSourceConfig:
+    """
+    Configuration of a CSV data source.
+    """
+
+    delimiter: str
+    has_headers: bool
+
+    def __init__(self, delimiter: str, has_headers: bool): ...
+
+class JsonSourceConfig:
+    """
+    Configuration of a JSON data source.
+    """
+
+class FileFormatConfig:
+    """
+    Configuration for parsing a particular file format (Parquet, CSV, JSON).
+    """
+
+    config: ParquetSourceConfig | CsvSourceConfig | JsonSourceConfig
+
+    @staticmethod
+    def from_parquet_config(config: ParquetSourceConfig) -> FileFormatConfig:
+        """
+        Create a Parquet file format config.
+        """
+        ...
+    @staticmethod
+    def from_csv_config(config: CsvSourceConfig) -> FileFormatConfig:
+        """
+        Create a CSV file format config.
+        """
+        ...
+    @staticmethod
+    def from_json_config(config: JsonSourceConfig) -> FileFormatConfig:
+        """
+        Create a JSON file format config.
+        """
+        ...
+    def file_format(self) -> FileFormat:
+        """
+        Get the file format for this config.
+        """
+        ...
+    def __eq__(self, other: FileFormatConfig) -> bool: ...  # type: ignore[override]
+    def __ne__(self, other: FileFormatConfig) -> bool: ...  # type: ignore[override]
+
+class FileInfo:
+    """
+    Metadata for a single file.
+    """
+
+    file_path: str
+    file_size: int | None
+    num_rows: int | None
+
+class FileInfos:
+    """
+    Metadata for a collection of files.
+    """
+
+    file_paths: list[str]
+    file_sizes: list[int | None]
+    num_rows: list[int | None]
+
+    @staticmethod
+    def from_infos(file_paths: list[str], file_sizes: list[int | None], num_rows: list[int | None]) -> FileInfos: ...
+    @staticmethod
+    def from_table(table: PyTable) -> FileInfos:
+        """
+        Create from a Daft table with "path", "size", and "num_rows" columns.
+        """
+        ...
+    def extend(self, new_infos: FileInfos) -> FileInfos:
+        """
+        Concatenate two FileInfos together.
+        """
+        ...
+    def __getitem__(self, idx: int) -> FileInfo: ...
+    def to_table(self) -> PyTable:
+        """
+        Convert to a Daft table with "path", "size", and "num_rows" columns.
+        """
+    def __len__(self) -> int: ...
+
+class S3Config:
```

Comment:
Should we also fill out the `__init__` for `S3Config`, `AzureConfig`, `GCSConfig`

Reply:
Good catch! I'll add those.